lms/djangoapps/courseware/models.py

```diff
@@ -485,3 +485,18 @@ class Meta:
         default=False,
         help_text=_('Disable the dynamic upgrade deadline for this organization.')
     )
+
+
+class LastSeenCoursewareTimezone(models.Model):
+    """
+    The timezone in the user's account is frequently not set.
+    This model stores a user's recent timezone that can be used as a fallback
+
+    .. no_pii:
+    """
+
+    user = models.ForeignKey(User, db_index=True, on_delete=models.CASCADE)
```

Comment:
why not use a OneToOneField here?  Would we ever need multiple LastSeenCoursewareTimezone rows for a single user?

Reply:
hm didn't think of that, fixed